### PR TITLE
Add initial project stubs, CI, tests and first example

### PR DESCRIPTION
**Pull request by ml-evs**
cc @yue-here and @jdbocarsly. Feel free to expand on any of this in a future PR, but this should be enough to get us up and running!

I had to pass my `ANTHROPIC_API_KEY` separately through my environment for the example notebook to work (it didn't seem to register when I added it through the Jupynaught(?) sidebar, so straight away this is something a `%%yellowhammer` cell magic could handle, or just some kind of `Yellowhammer` class that you interact with directly (perhaps more transferrable).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,37 @@
...
+
+### Launching example notebooks
+
+You can launch the example notebook locally with `uv` too:
+
+```shell
+uv run jupyter lab examples/
+```
+
+The examples will require you to bring your own *datalab* API key (for your instance of choice) and API keys for any underlying LLM providers (OpenAI, Anthropic, etc.).
```
Comment (jdbocarsly) [inline README.md:38]:
````suggestion
The examples will require you to bring your own *datalab* API key (for your instance of choice) and API keys for any underlying LLM providers (OpenAI, Anthropic, etc.).

These can be set in your shell profile, or simply in your shell before launching the jupyter notebook using:
```bash
export OPENAI_API_KEY=sk-proj...
export ANTHROPIC_API_KEY=sk-ant...
```

````

File: README.md [hunk 2]
```diff
@@ -4,3 +4,37 @@
...
+cd yellowhammer
+uv sync --dev
+```
+
+### Launching example notebooks
+
+You can launch the example notebook locally with `uv` too:
+
+```shell
+uv run jupyter lab examples/
```
Comment (jdbocarsly) [inline README.md:35]: This works for me, but `uv run jupyter notebook examples/` doesn't load the environment correctly... any idea why?

**Review by jdbocarsly — APPROVED** [review]
Thanks for setting this up!! Approved, but put a question and a comment

File: README.md (hunk 2 above)
Comment (ml-evs) [reply README.md:35]: `notebook` isn't installed by default, perhaps its pulling from somewhere else? I can add it to the lockfile

**Pull request merged**